sys/net/application_layer/gcoap/fileserver.c

```diff
@@ -21,25 +21,44 @@
 #include <unistd.h>
 #include <fcntl.h>
 
+#include "kernel_defines.h"
 #include "checksum/fletcher32.h"
 #include "net/gcoap/fileserver.h"
 #include "net/gcoap.h"
 #include "vfs.h"
+#include "vfs_util.h"
 
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
 /** Maximum length of an expressible path, including the trailing 0 character. */
 #define COAPFILESERVER_PATH_MAX (64)
 
+/** Randomly generated Etag, used by a client that a directory should only be
+    deleted, if it is empty */
+#define COAPFILESERVER_DIR_DELETE_ETAG (0x6ce88b56u)
```

Comment:
If a client is supposed to use this, shouldn't this be made available in a header?

Reply:
Also (sorry for jumping in so late, if this was discussed before), why is an ETag the signifier for this? Usually, with a REST API, this is done with some specific parameters (either as a query parameter or e.g. as a CBOR/JSON object).

Reply:
Ah, I saw that this was proposed by @chrysn in https://github.com/RIOT-OS/RIOT/pull/18133#discussion_r884165690. But still, I am a bit confused about this

Reply:
> But still, I am a bit confused about this

The content of an `If-Match` option is an Etag.
Simply formulated: If the client sends that special Etag, it expects the directory to be empty. And if so, the request is processed, which means it is `DELETE`ed. My first idea was indeed a query parameter, but I think the `If-Match` option works as well. Does this help or what exactly is unclear?